bool reverse
float32[] coef_x
float32[] coef_y
float32 duration